### ISSUE TITLE
Inconsistency in core's event systems, SceneJS_eventManager vs SceneJS_events.addListener vs SceneJS.Node.prototype.addListener

### ISSUE THREAD

**Issue by arjamizo** [xeolabs/scenejs#332]
**tl;dr** there is no way to inform external module that scene was created. Shall I allow doing that? 

I am working on SceneJS implementation, which would notify higher level (plugin system in this case), that it initialized correctly. 
In order to achieve that I am using `SceneJS_events.addListener` method, which is marked as `@private`

```
/**
 *  @private
 */
var SceneJS_events = new (function () {


    this.ERROR = 0;
    this.RESET = 1;                         // SceneJS framework reset
    this.NODE_CREATED = 2;                 // Scene has just been created
    this.SCENE_CREATED = 3;                 // Scene has just been created
    this.SCENE_COMPILING = 4;               // Scene about to be compiled and drawn
    this.SCENE_DESTROYED = 5;               // Scene just been destroyed
    this.OBJECT_COMPILING = 6;
    this.WEBGL_CONTEXT_LOST = 7;
    this.WEBGL_CONTEXT_RESTORED = 8;

    this.addListener = function (type, command, priority) {
```

I would like to know why event system is split into two subsystems: 

First, which is used only internaly and expose API for few cases, see `switch`, of course it is marked as `@deprecated`, but event carrying information about creation of the scene is executed only using [this API](https://github.com/xeolabs/scenejs/blob/b1930f6c35aac5fdb70ab591cc4d09533ef7cc38/src/core/events.js#L119-L161): 

```
/**
 * Subscribe to SceneJS events
 * @deprecated
 */
SceneJS.bind = function (name, func) {
    switch (name) {

        case "error" :

        case "reset" :

        case "webglcontextlost" :

        case "webglcontextrestored" :

        default:
```

and [second one](https://github.com/xeolabs/scenejs/blob/b1930f6c35aac5fdb70ab591cc4d09533ef7cc38/src/core/scene/node.js#L770-L783), which is used by nodes (_here I see the purpose, each node could have its own MPI aka events or msg system or pubsub system_)

```
SceneJS.Node.prototype.addListener = function (eventName, fn, options) {
    var list = this._listeners[eventName];
    if (!list) {
        list = [];
        this._listeners[eventName] = list;
    }
    list.push({
        eventName:eventName,
        fn:fn,
        options:options || {}
    });
    this._numListeners++;
    return this;
};
```

I have also found [this one](https://github.com/xeolabs/scenejs/blob/b1930f6c35aac5fdb70ab591cc4d09533ef7cc38/src/core/eventManager.js)

```
/**
 *  @private
 */
var SceneJS_eventManager = function () {

    this._handlerIds = new SceneJS_Map();

    this.typeHandlers = {};
};
```

Not to leave this issue without any point, I would like to know **how to notify external system that scene was created?** For now I am doing: 

```
SceneJS_events.addListener(SceneJS_events.SCENE_CREATED, return function(e) {
    console.info('loaded scene! from now on you can add new nodes to the view', e);
});
```

but since I am switching to RequireJS AMD solution, I cannot obtain instance of `SceneJS_events` (due to its privateness) anymore. 

**Solution from my point of view**

I am looking forward API like (which I am willing to implement if you do not mind): 

```
this.createScene = function (json, options) {
```

...

```
-        SceneJS_events.fireEvent(SceneJS_events.SCENE_CREATED, {    // Notify modules that need to know about new scene
-            engine:engine
-        });
+    SceneJS.Events.fire('scene-created', {engine: engine})
```
